Bottom: a9e0fdfc9375c1afd02e5593aee3038d1395c0ca
Top:    ff67cbbb0c19006e1719e3fad4ff9a024fee1766
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:57:50 +0000

polygon.c: Refactor NoHolesPolygonDicer to call poly_Copy0 rather than poly_Copy1

poly_Copy1 is much more of an internal function to polygon1.c, and this is its
only external caller. Replace this occurance with poly_Copy0, which is intended
to be the public API.


---

diff --git a/src/polygon.c b/src/polygon.c
index b90db0f..acdab30 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -2011,9 +2011,8 @@ NoHolesPolygonDicer (PolygonType *p, const BoxType * clip,
 {
   POLYAREA *main_contour, *cur, *next;
 
-  main_contour = poly_Create ();
   /* copy the main poly only */
-  poly_Copy1 (main_contour, p->Clipped);
+  poly_Copy0 (&main_contour, p->Clipped);
   /* clip to the bounding box */
   if (clip)
     {
